### PR TITLE
Use test specific directory in test_remote_extensions

### PR DESCRIPTION
**Pull request by lubennikovaav**
## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: test_runner/regress/test_download_extensions.py [hunk 1]
```diff
@@ -104,7 +77,7 @@ def endpoint_handler_build_tag(request: Request) -> Response:
 
     # Start a compute node with remote_extension spec
     # and check that it can download the extensions and use them to CREATE EXTENSION.
-    env = neon_env_builder.init_start()
+    env = neon_env_builder_local.init_start()
     tenant_id, _ = env.neon_cli.create_tenant()
```
Comment (koivunej) [inline test_runner/regress/test_download_extensions.py:81]: Drive-by: there is already the initial tenant and timeline created.

**Review by koivunej — APPROVED** [review]
To me this looks like it should work, and what we were aiming for earlier with @bayandin with the post-#4948 work.

File: test_runner/regress/test_download_extensions.py (hunk 1 above)
Comment (koivunej) [reply test_runner/regress/test_download_extensions.py:81]: One could avoid creating them with `init_config`.

Comment (lubennikovaav) [reply test_runner/regress/test_download_extensions.py]: please, check latest commit
agree that extra tenant is not needed
